eng/targets/Java.Common.targets

```diff
@@ -0,0 +1,87 @@
+<Project DefaultTargets="Build">
+
+  <PropertyGroup>
+    <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)' == ''">$(MSBuildProjectDirectory)\obj\</BaseIntermediateOutputPath>
+    <IntermediateOutputPath>$([MSBuild]::NormalizeDirectory('$(BaseIntermediateOutputPath)'))$(Configuration)\</IntermediateOutputPath>
+    <BuildDependsOn>
+      PrepareForBuild;
+      ResolveProjectReferences;
+      _Build;
+    </BuildDependsOn>
+    <JavaBuildArgs Condition="'$(JavaBuildArgs)' == ''">../gradlew $(GradleOptions) compileJava</JavaBuildArgs>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <JavaFiles Include="src\**\*.java" />
+    <JavaFiles Include="*.javaproj" />
+    <JavaFiles Include="build.gradle" />
```

Comment:
Nice.

Curious: Why does a `*.javaproj` project need to include itself in this item group❔

Reply:
If the javaproj is modified then build should be out-of-date. Don't think that happens unless it's added as part of the hash